sklearn/ensemble/gradient_boosting.py

```diff
@@ -634,7 +646,8 @@ def _update_terminal_region(self, tree, terminal_regions, leaf, X, y,
         numerator = np.sum(y_ * sample_weight * np.exp(-y_ * pred))
         denominator = np.sum(sample_weight * np.exp(-y_ * pred))
 
-        if denominator == 0.0:
+        # prevents overflow and division by zero
+        if abs(denominator) < 1e-150:
```

Comment:
Hmm.. I am not sure. @raghavrv  do you have a strong preference for `np.finfo(np.float32).eps` or do you think `1e-150` is still fine? I personally prefer `1e-150` because the case where this algorithm was failing at was when `denominator` was around `1e-309`, so I felt that `1e-150` was appropriate since it's about half of 300. 

```
>>> np.finfo(np.float).eps
2.2204460492503131e-16
>>> np.finfo(np.double).eps
2.2204460492503131e-16
```

It's just that I am kind of worried that those values are too large compared to `1e-150` and I'm not sure if it will cause any rounding errors.